mne/io/cnt/cnt.py

```diff
@@ -1,26 +1,106 @@
 """Conversion tool from Neuroscan CNT to FIF."""
 
 # Author: Jaakko Leppakangas <jaeilepp@student.jyu.fi>
+#         Joan Massich <mailsik@gmail.com>
 #
 # License: BSD (3-clause)
 from os import path
-import datetime
-import calendar
 
 import numpy as np
 
 from ...utils import warn, verbose, fill_doc, _check_option
 from ...channels.layout import _topo_to_sphere
 from ..constants import FIFF
-from ..utils import _mult_cal_one, _find_channels, _create_chs, read_str
+from ..utils import (_mult_cal_one, _find_channels, _create_chs, read_str,
+                     _deprecate_stim_channel)
 from ..meas_info import _empty_info
 from ..base import BaseRaw, _check_update_montage
+from ...annotations import Annotations
+
+from struct import unpack, calcsize
+
+from ._utils import (_read_teeg, _get_event_parser, _session_date_2_meas_date,
+                     CNTEventType3)
+
+
+def _read_annotations_cnt(fname):
+    """CNT Annotation File Reader.
+
+    This method opens the .cnt files, searches all the metadata to construct
+    the annotations and parses the event table. Notice that CNT files, can
+    point to a different file containing the events. This case when the
+    event table is separated from the main .cnt is not supported.
+
+    Parameters
+    ----------
+    fname: str
+        path to cnt file containing the annotations.
+
+    Returns
+    -------
+    onset : array of float, shape (n_annotations,)
+        The starting time of annotations in seconds after ``orig_time``.
+    duration : array of float, shape (n_annotations,)
+        Durations of the annotations in seconds.
+    description : array of str, shape (n_annotations,)
+        Array of strings containing description for each annotation. If a
+        string, all the annotations are given the same description. To reject
+        epochs, use description starting with keyword 'bad'. See example above.
+    """
+    # Offsets from SETUP structure in http://paulbourke.net/dataformats/eeg/
+    SETUP_NCHANNELS_OFFSET = 370
+    SETUP_RATE_OFFSET = 376
+    SETUP_EVENTTABLEPOS_OFFSET = 886
+
+    def _translating_function(offset, n_channels, event_type, n_bytes=2):
+        # n_bytes is related to _get_cnt_info's data_format parameter
+        # 'auto', 'int16', and 'int32'
+        if event_type == CNTEventType3:
+            offset *= n_bytes * n_channels
+        event_time = offset - 900 - (75 * n_channels)
+        event_time //= n_channels * n_bytes
+        return event_time - 1
+
+    with open(fname, 'rb') as fid:
+        fid.seek(SETUP_NCHANNELS_OFFSET)
+        (n_channels,) = unpack('<H', fid.read(calcsize('<H')))
```

Comment:
typically (and more compactly) we use and prefer `np.frombuffer` instead of `fid.read` + `unpack`

Reply:
if `np.frombuffer` took a file descriptor (or the file was lodaded in a buffer) this could be written like this which is really readable:

 ```py
    with open(fname, 'rb') as fid:
        n_channels = np.frombuffer(
            fid, dtype=np.uint16, offset=SETUP_NCHANNELS_OFFSET)
        sfreq = np.frombuffer(
            fid, dtype=np.uint16, offset=SETUP_RATE_OFFSET)
        event_table_pos = np.frombuffer(
            fid, dtype=np.int32, offset=SETUP_EVENTTABLEPOS_OFFSET)
```

Otherwise `np.fromfile` but you can not get read of the seek.

```py
    with open(fname, 'rb') as fid:
        fid.seek(SETUP_NCHANNELS_OFFSET)
        n_channels = np.fromfile(fid, dtype='<u2', count=1)[0]

        fid.seek(SETUP_RATE_OFFSET)
        sfreq =  np.fromfile(fid, dtype='<u2', count=1)[0]

        fid.seek(SETUP_EVENTTABLEPOS_OFFSET)
        event_table_pos = np.fromfile(fid, dtype='<i4', count=1)[0]
```
But I've no idea which of the 3 is preferable.

Reply:
`frombuffer` is what we use in `mne/io/tag.py` and it works with a file descriptor, we changed it to that from `np.fromfile` a year or so ago so I think it's the preferred way

Reply:
(or rather I guess we do `frombuffer(fid.read(...), ...)` in `tag.py`)

Reply:
I would say do whichever works. Having a `fid.seek` in there is expected/fine